### PR TITLE
Use argparse for command argument parser for lfsccm

### PR DESCRIPTION
**Pull request by bloodeagle40234**
For testing perspective, C lang like getopt  would be complex to generate flexible configuration and the option syntax check. With those commits in this PR, the newer lfsccm is going to use python native argparse library and make all actions command to subargument to ensure the required options into the subcommands.

### CODE REVIEW

File: lfsccm/lfsccm/main.py [hunk 1]
```diff
@@ -304,46 +337,71 @@ def parse_arg(arg):
...
-            logger.error("Invalid option [{}]".format(opt))
-            usage()
-            return 1
+    parser = argparse.ArgumentParser()
+    sub_parser = parser.add_subparsers(
+        dest="actions",
+        help="Action verb for operation")
+    attach_parser = sub_parser.add_parser("attach")
+    # attach
+    for option in ["nodes", "roids", "rwids"]:
```
Comment (kaneko-tx) [inline lfsccm/lfsccm/main.py:346]: Attach requires --files option.

File: lfsccm/lfsccm/main.py [hunk 2]
```diff
@@ -304,46 +337,71 @@ def parse_arg(arg):
...
+        attach_parser.add_argument(
+            *AVAILABLE_OPTIONS[option]["kargs"],
+            **AVAILABLE_OPTIONS[option]["kwargs"])
+    # detach
+    detach_parser = sub_parser.add_parser("detach")
+    for option in ["nodes", "roids", "rwids"]:
```
Comment (kaneko-tx) [inline lfsccm/lfsccm/main.py:352]: Detach requires --files option.

File: lfsccm/lfsccm/main.py [hunk 3]
```diff
@@ -304,46 +337,71 @@ def parse_arg(arg):
...
 
-    action = args[0].lower()
+    action = args.actions
+    # FIXME: for safety set empty list if it's not required
+    nodes = []
+    ro_ids = []
+    rw_ids = []
+    files =  []
+    if hasattr(args, "nodes"):
+        ro_ids = args.nodes
```
Comment (kaneko-tx) [inline lfsccm/lfsccm/main.py:393]: Maybe `nodes` is correct, not `ro_ids`.

**Review by kaneko-tx — COMMENTED** [review]
Thank you for the modifications. I would like to make a few comments.

https://github.com/DDNStorage/lfsccm/blob/0aae1e5e0b09f08fe10277114d037923bc8538a3/lfsccm/lfsccm/main.py#L191-L201
getopt returns the number as str, but argparse returns it as int. So `isinstance(self.rw_id, int)` is fine instead of `self.rw_id.isdecimal()`. Same for ro_id.

https://github.com/DDNStorage/lfsccm/blob/0aae1e5e0b09f08fe10277114d037923bc8538a3/lfsccm/lfsccm/main.py#L331-L334
Maybe `parse_arg()` is no longer needed.

**Comment by bloodeagle40234** [comment]
Add fix commit addressing to @kaneko-tx 's comments and rebased to the conflicted main branch.

**Review by kaneko-tx — APPROVED** [review]
Thank you! LGTM.

**Review by rkmrHonjo — APPROVED** [review]
LGTM

**Pull request merged**